src/bio.c

```diff
@@ -134,7 +134,7 @@ static int wolfSSL_BIO_SSL_read(WOLFSSL_BIO* bio, void* buf,
     WOLFSSL_ENTER("wolfSSL_BIO_SSL_write");
 
     /* already got eof, again is error */
-    if (bio && front->eof)
+    if (front->eof)
```

Comment:
This probably should be `if (front && front->eof)`.

Reply:
Is the intention to avoid dereferencing "front" when it's NULL? Since this condition returns an error when true, it could be:
```    
if ((front == NULL) || front->eof)
        return WOLFSSL_FATAL_ERROR;
```

Reply:
This logic is hard to follow, but the `wolfSSL_BIO_read` API is an openssl compatibility one. If in doubt about functionality sometimes its helpful to go check the openssl sources. In this case the NULL check on front is to prevent a NULL dereference. If the front->eof is set then the read has reached EOF and it should not try and read again.